asvdb/asvdb.py

```diff
@@ -405,8 +405,10 @@ def __readResults(self, infoOnly=False, filterByInfoObjs=None):
                     if infoOnly:
                         machineResults.append(bi)
                     else:
-                        # FIXME: if results not in rDict, throw better error
-                        resultsDict = rDict["results"]
+                        resultsDict = rDict.get("results",{})
+                        if not resultsDict:
+                            print("ERROR: No results found!")
```

Comment:
should this raise an exception if it's an error?

Reply:
The question is, is it a stop error or a fallthrough and notify error.  I tend to only stop a program when the error will create incorrect perceptions or break other functionality.  If you think this is a stop case, I will change it to be an exception. To me it looked like an unlikely case, but not impossible.  I still raised the error message, because there shouldn't be a case where there is a key for the results but no stored results dict.